### DISCUSSION TITLE
How does a  insertNode transform itself?

### DISCUSSION THREAD

**Discussion by fovj** [L3MON4D3/LuaSnip#1102] [Q&A]
I want to automatically convert the content when editing after expanding the snippet.

However, it seems to require 2 nodes. 

```lua
s(
  "n",
  fmta(
    [[
  lower: <>
  upper: <>
  ]],
    {
      i(1, "text"), -- I don't need this node, 
      f(function(args) -- but the functionNode needs to obtain its real-time value.
        return string.upper(args[1][1])
      end, { 1 }),
    }
  )
),

```

This is close, but it cannot obtain real-time input content from the `args`.
```lua
s(
  "n",
  fmta(
    [[
  upper: <>
  ]],
    {
      f(function(args) -- {}
        return "abc"
      end),
    }
  )
),
```

**Comment by L3MON4D3** [comment]
Ah, that's currently a limitation :/
But I'm working on it, I'll update when I have something concrete :)

**Reply by L3MON4D3** [reply]
I got it working :)
Check #1137
